sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -32,65 +37,131 @@
  */
 @AutoValue
 public abstract class DoFnSignature {
+  /** Class of the original {@link DoFn} from which this signature was produced. */
   public abstract Class<? extends DoFn> fnClass();
 
+  /** Whether this {@link DoFn} does a bounded amount of work per element. */
+  public abstract PCollection.IsBounded isBounded();
+
+  /** Details about this {@link DoFn}'s {@link DoFn.ProcessElement} method. */
   public abstract ProcessElementMethod processElement();
 
+  /** Details about this {@link DoFn}'s {@link DoFn.StartBundle} method. */
   @Nullable
   public abstract BundleMethod startBundle();
 
+  /** Details about this {@link DoFn}'s {@link DoFn.FinishBundle} method. */
   @Nullable
   public abstract BundleMethod finishBundle();
 
+  /** Details about this {@link DoFn}'s {@link DoFn.Setup} method. */
   @Nullable
   public abstract LifecycleMethod setup();
 
+  /** Details about this {@link DoFn}'s {@link DoFn.Teardown} method. */
   @Nullable
   public abstract LifecycleMethod teardown();
 
+  /** Details about this {@link DoFn}'s {@link DoFn.GetInitialRestriction} method. */
+  @Nullable
+  public abstract GetInitialRestrictionMethod getInitialRestriction();
+
+  /** Details about this {@link DoFn}'s {@link DoFn.SplitRestriction} method. */
+  @Nullable
+  public abstract SplitRestrictionMethod getSplitRestriction();
+
+  /** Details about this {@link DoFn}'s {@link DoFn.GetRestrictionCoder} method. */
+  @Nullable
+  public abstract GetRestrictionCoderMethod getRestrictionCoder();
+
+  /** Details about this {@link DoFn}'s {@link DoFn.NewTracker} method. */
+  @Nullable
+  public abstract NewTrackerMethod newTracker();
+
   static DoFnSignature create(
```

Comment:
In this or a follow-up CL, I would strongly suggest switching DoFnSignature to using a builder rather than a single create method. This is getting pretty long, and especially since it has multiple nullable fields, etc., it would be really easy for us to mix them up with odd consequences.